clients/cli/cmd/internal/push.go

```diff
@@ -209,13 +212,29 @@ func (source *Source) Push(client *phrase.APIClient, waitForResults bool, branch
 			fmt.Fprintln(os.Stderr, strings.Repeat("-", 10))
 		}
 	}
-	if !noErrors {
+	if noErrors {
+		if waitForResults && cleanup {
+			cleanupAfterUpload(client, source.ProjectID, uploadIds)
+		}
+	} else {
 		return errors.New("not all files were uploaded successfully")
 	}
 
 	return nil
 }
 
+func cleanupAfterUpload(client *phrase.APIClient, projectId string, uploadIds []string) error {
+	cleanupCommand := &UploadCleanupCommand{
```

Comment:
How about putting the logic for deleting the keys into this method and then call it from the clean up command? Simulating a command to cleanup the keys feels a bit like the wrong way around